option.go

```diff
@@ -10,13 +15,257 @@ type option interface {
 	GetSource() string
 }
 
+// StringifyValue is global variable to indicate if the Option should be
+// serialized as just the value (when value is true) or if the entire Option
+// struct should be serialized.  This is a hack, and not recommended for general
+// usage, but can be useful for debugging.
 var StringifyValue = true
 
-// used in option parsing for map types Set routines
+// stringMapRegex is used in option parsing for map types Set routines
 var stringMapRegex = regexp.MustCompile("[:=]")
 
+type Option[T any] struct {
+	Source  string
+	Defined bool
+	Value   T
+}
+
+func NewOption[T any](dflt T) Option[T] {
+	return Option[T]{
+		Source:  "default",
+		Defined: true,
+		Value:   dflt,
+	}
+}
+
+func (o Option[T]) IsDefined() bool {
+	return o.Defined
+}
+
+func (o *Option[T]) SetSource(source string) {
+	o.Source = source
+}
+
+func (o *Option[T]) GetSource() string {
+	return o.Source
+}
+
+func (o Option[T]) GetValue() interface{} {
+	return o.Value
+}
+
+// This is useful with kingpin option parser
+func (o *Option[T]) Set(s string) error {
+	err := convertString(s, &o.Value)
+	if err != nil {
+		return err
+	}
+	o.Source = "override"
+	o.Defined = true
+	return nil
+}
+
+// This is useful with survey prompting library
+func (o *Option[T]) WriteAnswer(name string, value interface{}) error {
+	if v, ok := value.(T); ok {
+		o.Value = v
+		o.Defined = true
+		o.Source = "prompt"
+		return nil
+	}
+	return fmt.Errorf("Got %T expected %T type: %v", value, o.Value, value)
+}
+
+func (o *Option[T]) SetValue(v interface{}) error {
```

Comment:
Same; curious why this isn't strongly typed. Is it because kingpin needs to cast the `Option` to an interface that defines `SetValue(v interface{})`?

Reply:
Yeah, kingpin defined this interface so we have to implement.